### ISSUE TITLE
FQN -> import can add a duplicated import

### ISSUE THREAD

**Issue by przepompownia** [phpactor/phpactor#1941]
If we try to convert inline `Z\X` to import, we get a duplicate `use Z\X` if there exist another an import for `X`, e.g. `use A\X`.

Thanks for #1939 anyway!

**Comment by mamazu** [comment]
Thanks for reporting the bug. I actually thought about that but forget to add a test for it. But working on it now.

**Comment by mamazu** [comment]
@dantleech I would suggest making the `NameImporter` logic more modular so that it can also be here. What so you think?

**Comment by dantleech** [comment]
hm, this works for me:

![recorded](https://user-images.githubusercontent.com/530801/201194098-16b800d1-2bd7-4437-b4cc-dcde68d927ae.gif)

**Comment by mamazu** [comment]
The example I came up with is the following:
```
<?php

use SomeOtherNamespace\Foo;

function (\SomeNamespace\Foo<> $foo): void {
}
```

Running the command just blindly creates another use statment with the same name like so:

```
<?php

use SomeOtherNamespace\Foo;
use SomeNamespace\Foo;

function (Foo $foo): void {
}
```
Which obviously doesn't work.

**Comment by dantleech** [comment]
ok, so yes there is some logic to automatically alias name imports for auto-complete etc, which is missing from refactorings (not just this one). So this is not a problem with this feature, but with the "Updater" in general, you would I guess get the same issue when implementing contracts for example when a name is already imported, or extracting a method.

**Comment by mamazu** [comment]
Yeah but you already have solved it in the `ImportName` command. Wouldn't it be possible and useful to extract that functionality out and use it everywhere where use statements are generated?

**Comment by dantleech** [comment]
it may make sense to refactor to enable the Updater to use the same logic
